src/extent.c

```diff
@@ -812,7 +851,19 @@ extent_recycle_extract(tsdn_t *tsdn, arena_t *arena,
 			extent_unlock(tsdn, unlock_extent);
 		}
 	} else {
-		extent = extents_fit_locked(tsdn, arena, extents, alloc_size);
+		extent = NULL;
```

Comment:
This one actually makes the logic much trickier to follow, since both !slab and sized alloc region failure can cause the fallback to be necessary.